lib/component.hpp

```diff
@@ -1,24 +1,55 @@
 #ifndef COMPONENT_HPP
 #define COMPONENT_HPP
 
+#include "messages/spa_subscription_reply.hpp"
+#include "messages/spa_subscription_request.hpp"
 #include "spa_communicator.hpp"
+#include "spa_message.hpp"
+#include <iostream>
 #include <memory>
 
 class Component
 {
 public:
   typedef std::shared_ptr<SpaCommunicator> Com;
 
-  Component(Com communicator = nullptr) : communicator(communicator) {}
-  // virtual ~Component(){}
+  Component(Com communicator = nullptr, LogicalAddress address = LogicalAddress(0, 0)) : communicator(communicator),
+                                                                                         address(address),
+                                                                                         dialogId(0) {}
+  virtual ~Component() {}
   virtual void appInit() = 0;
   // virtual void appShutdown();
   // virtual int32_t publish();
-  // virtual void registerRequest();
-  virtual void sendMsg(std::shared_ptr<SpaMessage> message);
-  // virtual void receiveMessage(SpaMessage);
+
+  template <typename M>
+  void sendMsg(M);
```

Comment:
I found that with this function specifically, a template works just fine. It makes it so we have one less cast to do.